packages/cli/src/constructs/check.ts

```diff
@@ -16,90 +16,215 @@ import { ConfigDefaultsGetter, makeConfigDefaultsGetter } from './check-config'
    * @deprecated Use {@link retryStrategy} instead.
    */
   doubleCheck?: boolean
+  
   /**
-   * Allows to invert the behaviour of when a check is considered to fail. Allows for validating error status like 404.
-   * This only applies to API Checks.
+   * Allows to invert the behaviour of when a check is considered to fail. 
+   * Useful for validating error status codes like 404 or 500.
+   * This only applies to API Checks. When set to true, the check passes when 
+   * it would normally fail, and fails when it would normally pass.
+   * 
+   * @defaultValue false
+   * @example
+   * ```typescript
+   * // Check that expects a 404 status - must set shouldFail: true
+   * shouldFail: true,
+   * request: {
+   *   method: 'GET',
+   *   url: 'https://api.example.com/nonexistent',
+   *   assertions: [AssertionBuilder.statusCode().equals(404)]
+   * }
+   * ```
    */
   shouldFail?: boolean
+  
   /**
-   * An array of one or more data center locations where to run this check. The supported regions are:
+   * An array of one or more data center locations where to run this check.
+   * The supported regions are:
    * us-east-1, us-east-2, us-west-1, us-west-2, ca-central-1, sa-east-1,
    * eu-west-1, eu-central-1, eu-west-2, eu-west-3, eu-north-1, eu-south-1, me-south-1,
    * ap-southeast-1, ap-northeast-1, ap-east-1, ap-southeast-2, ap-southeast-3, ap-northeast-2, ap-northeast-3,
    * ap-south-1, af-south-1
+   * 
+   * @example ['us-east-1', 'eu-west-1', 'ap-southeast-1']
+   * @see {@link https://www.checklyhq.com/docs/monitoring/global-locations/ | Global Locations}
    */
   locations?: Array<keyof Region>
+  
   /**
    * An array of one or more private locations where to run the check.
    * PrivateLocation instances or slug name strings are allowed.
    *
    * `string` slug names are **only** allowed for private locations that **not** belong to the project. Use
    * PrivateLocation instances references for private locations created within the project.
+   * 
+   * @example
+   * ```typescript
+   * // Using private location instances
+   * privateLocations: [myPrivateLocation, anotherPrivateLocation]
+   * 
+   * // Using existing private location slugs
+   * privateLocations: ['my-datacenter-1', 'office-location']
+   * ```
    */
   privateLocations?: Array<string|PrivateLocation|PrivateLocationRef>
+  
   /**
-   * Tags for organizing and filtering checks.
+   * Tags for organizing and filtering checks in the dashboard.
+   * 
+   * @example ['production', 'api', 'critical']
    */
   tags?: Array<string>
+  
   /**
-   * How often the check should run in minutes.
+   * How often the check should run. Can be specified in minutes or using Frequency constants.
+   * 
+   * @example
+   * ```typescript
+   * frequency: Frequency.EVERY_5M  // Using frequency constant
+   * frequency: 10                   // Every 10 minutes
+   * ```
    */
   frequency?: number | Frequency
```

Comment:
Is this correct? Do we allow custom frequencies?

Reply:
No, we can't introduce breaking changes in a docs PR.

Reply:
Updated